tower/tests/builder.rs

```diff
@@ -1,69 +1,52 @@
-use futures::{
-    future::{self, FutureResult},
-    prelude::*,
-};
 use std::time::Duration;
+use futures_util::{pin_mut, future::{Ready, poll_fn}};
 use tower::builder::ServiceBuilder;
+use tower::util::ServiceExt;
 use tower_buffer::BufferLayer;
 use tower_limit::{concurrency::ConcurrencyLimitLayer, rate::RateLimitLayer};
 use tower_retry::{Policy, RetryLayer};
 use tower_service::*;
-use void::Void;
+use tower_test::{mock};
+
+#[tokio::test]
```

Comment:
block_on should work though not sure if its preferred.

Reply:
It's actually pretty annoying to convert this to a non `#[tokio::test]`, since buffer ends up spawning a worker, which gets awkward in mock context (and for some reason also with `block_on`).